ui/raidboss/popup-text.js

```diff
@@ -426,6 +426,11 @@ class PopupText {
     let suppress = 'suppressSeconds' in trigger ? ValueOrFunction(trigger.suppressSeconds) : 0;
     if (trigger.id && suppress > 0)
       this.triggerSuppress[trigger.id] = now + suppress * 1000;
+    let promise = 'promise' in trigger ? promise : () => {
+      return new Promise((res) => {
+        res(matches);
```

Comment:
My inclination is to push back on this, because I'm not sure I follow what the use case is for modifying matches.  Can you help me understand?

If you need to return something from the promise, maybe there could be a better way, like calling functions with (data, matches, promiseReturn)?

Reply:
Maybe for this case the `trigger.promise` function should accept either the `this.data` attribute or an entirely separate `promiseData` local object. The latter would require modifying the rest of the trigger code inside the local `f` function to account for the new variable, though. If you're disinclined to replacing the `matches` object then probably the former (pass in `this.data` to `trigger.promise`) is the best option.

Reply:
Sorry, it just feels a little inconsistent from the way other trigger functions work.  Right now, passing data between trigger functions is by modifying the data param (which is why ordering is important too).  Modifying matches to return data (and knowing that if you return anything that it will clobber the matches variable) seems a bit fragile too.

Calling `trigger.promise(data, matches)` and having the promise modify `data` (like every other trigger function) to communicate with other trigger functions seems much more consistent/intuitive (even if it's kind of clunky at the same time, admittedly).  And then I guess in that world the promise function just doesn't return anything.

Reply:
Alright, I'll make these adjustments. Not sure when I'll get the chance to add examples/documentation, might be mid-next-week.